src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(a) {
+    if (a === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += a;
+
+    return adder;
```

Comment:
The function `adder` correctly accumulates the sum and resets it when called with no arguments. However, it's important to ensure that the function cannot be misused by passing non-numeric values, which could lead to incorrect sum calculations. While the task specifies not to check that the argument is a number, it is still a good practice to be aware of such potential issues in real-world scenarios.